### PR TITLE
Trace attention inference patterns with p=0, cleanup

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109066
* #109156
* #109142
* #108917
* #108894
* __->__ #109118

When dropout is traced in inference, it creates a clone() instead of training pattern of rand() etc. This was partially addressed by manually https://github.com/pytorch/pytorch/pull/108141, however that did not cover all of the patterns that included dropout, and there is no reason we should have to specify them manually.

This updates the inference patterns generated to trace with dropout_p = 0.0.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: test/inductor/test_fused_attention.py [hunk 1]
```diff
@@ -50,20 +52,24 @@ def _check_common(
                 torch.randn(tensor_shape, device=self.device),
                 torch.randn(tensor_shape, device=self.device),
             ]
+        else:
+            args1 = list(args1)
         args2 = self._clone_inputs(args1)
 
-        for training in [False, True]:
+        for training in [False] + ([True] if check_train else []):
```
Comment (drisspg) [inline test/inductor/test_fused_attention.py:59]: nit `[False, Train] if check_train else [False]`

**Review by drisspg — APPROVED** [review]
The actual pattern definitions look much cleaner!

**Review by Valentine233 — APPROVED** [review]
